main.cpp

```diff
@@ -73,418 +72,232 @@ compile/link/run to make sure you don't have any errors
         }
 
-    
-    private:
-        int* heapInt = nullptr;
-        IntType& powInternal( int x );
-};
-
-//float type operators
-
-
-//point constructor implementations
-Point::Point(FloatType& a, FloatType& b) : Point(static_cast<float>(a), static_cast<float>(b) ) { } 
-
-Point::Point(DoubleType& a, DoubleType& b) : Point ( static_cast<float>(a), static_cast<float>(b) ) { }
-
-
-Point::Point(IntType& a, IntType& b) : Point ( static_cast<float>(a), static_cast<float>(b) ) { }
-
-//point multiply UDT implementations
-Point& Point::multiply( FloatType& m )
-{
-    return multiply(static_cast<float>(m));
-}
-
-Point& Point::multiply( DoubleType& m )
-{
-    return multiply(static_cast<float>(m));
-}
-
-Point& Point::multiply( IntType& m )
-{
-    return multiply(static_cast<float>(m));
-}
-
-// point to string
-#include <iostream>
-void Point::toString()
-{
-    std::cout << "( " << x << ", " << y << " )" << std::endl;
-}
-
-// pow implementations for FloatType
-FloatType& FloatType::pow( const FloatType& operand )
-{
-    return powInternal( static_cast<float>(operand) );
-}
-
-FloatType& FloatType::pow( const DoubleType& operand )
-{
-    return powInternal( static_cast<float>(operand) );
-}
-
-FloatType& FloatType::pow( const IntType& operand )
-{
-    return powInternal( static_cast<float>(operand) );
-}
-
-FloatType& FloatType::pow( float operand )
-{
-    return powInternal(operand);
-}
-
-// pow implementations for DoubleType
-DoubleType& DoubleType::pow( const FloatType& operand )
-{
-    return powInternal( static_cast<double>(operand) );
-}
-
-DoubleType& DoubleType::pow( const DoubleType& operand )
-{
-    return powInternal( static_cast<double>(operand) );
-}
-
-DoubleType& DoubleType::pow( const IntType& operand )
-{
-    return powInternal( static_cast<double>(operand) );
-}
-
-DoubleType& DoubleType::pow( double operand )
-{
-    return powInternal(operand);
-}
-
-// pow implementations for IntType
-IntType& IntType::pow( const FloatType& operand )
-{
-    return powInternal( static_cast<int>(operand) );
-}
-
-IntType& IntType::pow( const DoubleType& operand )
-{
-    return powInternal( static_cast<int>(operand) );
-}
-
-IntType& IntType::pow( const IntType& operand )
-{
-    return powInternal( static_cast<int>(operand) );
-}
-
-IntType& IntType::pow( int operand )
-{
-    return powInternal(operand);
-}
-
+        *heapNumber /= y;
+        return *this;
+    }
 
-// powInternal implementations
-FloatType& FloatType::powInternal( float x )
-{
-    if( heapFloat != nullptr )
+    Numeric& pow( const Numeric& operand )
     {
-        *heapFloat = std::pow( *heapFloat, x );
+        return powInternal( operand );
     }
-    return *this;
-}
 
-DoubleType& DoubleType::powInternal( double x )
-{
-    if( heapDub != nullptr )
+    template<typename Callable>
+    auto apply(Callable callable) ->  Numeric&
     {
-        *heapDub = std::pow( *heapDub, x );
+        callable(*heapNumber);
+        return *this;
     }
```

Comment:
Your `apply` function should take your unique_ptr, not the unique_ptr's held object
see instruction 4)
- [ ] revise